File: backend/internal/kanban/presence.go
// Owner: Track 2 (Person B — WebSocket layer)
// Responsible for: presence tracking (who's online) and reconnection handling, both required by the mandatory multi-user support requirement.
package kanban

// BroadcastPresence sends a "joined"/"left" event to a project's room on client connect/disconnect.
func (h *Hub) BroadcastPresence(projectID uint, userID uint, status string) {
	// TODO: status is "joined" or "left"; broadcast via BroadcastToProject so presence indicators update live
}

// HandleReconnect re-syncs a client's state after a dropped-then-restored WebSocket connection.
func (h *Hub) HandleReconnect(client *Client) {
	// TODO: on reconnect, don't assume the socket resumed cleanly — have the client re-fetch current board state over HTTP
	// TODO: re-register the client and re-broadcast presence
}

File: frontend/src/kanban/presence.ts
// Owner: Track 2 (Person B — WebSocket layer)
// Responsible for: presence indicators (who's currently viewing this project) and reconnection handling, on top of Socket.IO.

// handlePresenceEvent updates the presence indicator UI on an incoming "joined"/"left" event.
function handlePresenceEvent(event: { userId: string; status: "joined" | "left" }): void {
  // TODO: add/remove the user from the visible "online now" avatar list
}

// handleReconnect re-fetches current board state after the Socket.IO client's "connect" event fires again, rather than assuming it resumed cleanly.
async function handleReconnect(projectId: string): Promise<void> {
  // TODO: hook this into the socket's "connect" event (fired again on Socket.IO's automatic reconnect)
  // TODO: re-fetch the board via boardApi.fetchBoard and reconcile local state
  // TODO: re-emit "join_project" to rejoin the project's Socket.IO room
}
